### PR TITLE
build: add conventional changelog support

### PR DESCRIPTION
**Pull request by kiaking**
This PR adds Conventional Changelog support to enhance release process and to align with recent Vue ecosystems.

- Generate initial CHANGELOG.md including all past releases.
- Add Contribution guideline taken from Vue & Vue next. The contents are modified slightly to fit into Vuex context.
- Update release shell file to support changeling generation.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,289 @@
+# [5.0.0](https://github.com/vuejs/vuex/compare/v3.1.3...v5.0.0) (2020-04-02)
```
Comment (ktsn) [inline CHANGELOG.md:1]: 3.2.0?

File: build/release.sh [hunk 2]
```diff
@@ -14,13 +14,21 @@ then
...
+  git commit -m "realese: v$VERSION"
+  git tag "v$VERSION"
 
   # publish
-  git push origin refs/tags/v$VERSION
-  git push
-  npm publish
+  # git push origin refs/tags/v$VERSION
+  # git push
+  # npm publish
```
Comment (ktsn) [inline build/release.sh:33]: Should we comment out these lines?

File: CHANGELOG.md (hunk 1 above)
Comment (kiaking) [reply CHANGELOG.md:1]: Oops sorry. Gonna fix it.

File: build/release.sh (hunk 2 above)
Comment (kiaking) [reply build/release.sh:33]: Oh no it shouldn't. I'll fix this.

**Comment by kiaking** [comment]
@ktsn Done fixing 👍

**Pull request merged**